test/e2e-monitoring/monitoring_test.go

```diff
@@ -81,15 +117,19 @@ func TestResourceVersionApplication(t *testing.T) {
 	if err != nil && !errors.IsAlreadyExists(err) {
 		t.Fatalf("Failed to create resource: %v", err)
 	}
-	require.True(t, didUpdate)
+	expectation, err := didUpdate(oldUnstructuredResource, gotUnstructuredResource)
+	if err != nil {
+		t.Fatalf("Failed to compare resources: %v", err)
+	}
+	require.True(t, expectation != noUpdate)
 
 	// Update the resource version and the generation since we made a spec change.
-	unstructuredResource.SetResourceVersion(gotUnstructured.GetResourceVersion())
-	unstructuredResource.SetGeneration(gotUnstructured.GetGeneration())
-	unstructuredResource.SetCreationTimestamp(gotUnstructured.GetCreationTimestamp())
-	unstructuredResource.SetUID(gotUnstructured.GetUID())
-	unstructuredResource.SetManagedFields(gotUnstructured.GetManagedFields())
-	require.Equal(t, unstructuredResource.UnstructuredContent(), gotUnstructured.UnstructuredContent())
+	unstructuredResource.SetResourceVersion(gotUnstructuredResource.GetResourceVersion())
+	unstructuredResource.SetGeneration(gotUnstructuredResource.GetGeneration())
+	unstructuredResource.SetCreationTimestamp(gotUnstructuredResource.GetCreationTimestamp())
+	unstructuredResource.SetUID(gotUnstructuredResource.GetUID())
+	unstructuredResource.SetManagedFields(gotUnstructuredResource.GetManagedFields())
+	require.Equal(t, unstructuredResource.UnstructuredContent(), gotUnstructuredResource.UnstructuredContent())
```

Comment:
Since we are already doing this now, LMK if the `didUpdate` function above feels redundant now. I kept it to visualize the workaround for the `didUpdate` return parameter in the test.